volume/volume_windows.go

```diff
@@ -94,109 +102,38 @@ func (m *MountPoint) BackwardsCompatible() bool {
+	var split []string
 	matchgroups := make(map[string]string)
+	// Pull out the sub expressions from the named capture groups
 	for i, name := range specExp.SubexpNames() {
 		matchgroups[name] = strings.ToLower(match[i])
 	}
-
-	mp := &MountPoint{
-		Source:      matchgroups["source"],
-		Destination: matchgroups["destination"],
-		RW:          true,
-	}
-	if strings.ToLower(matchgroups["mode"]) == "ro" {
-		mp.RW = false
-	}
-
-	// Volumes cannot include an explicitly supplied mode eg c:\path:rw
-	if mp.Source == "" && mp.Destination != "" && matchgroups["mode"] != "" {
-		return nil, errInvalidSpec(spec)
-	}
-
-	// Note: No need to check if destination is absolute as it must be by
-	// definition of matching the regex.
-
-	if filepath.VolumeName(mp.Destination) == mp.Destination {
-		// Ensure the destination path, if a drive letter, is not the c drive
-		if strings.ToLower(mp.Destination) == "c:" {
-			return nil, fmt.Errorf("Destination drive letter in '%s' cannot be c:", spec)
-		}
-	} else {
-		// So we know the destination is a path, not drive letter. Clean it up.
-		mp.Destination = filepath.Clean(mp.Destination)
-		// Ensure the destination path, if a path, is not the c root directory
-		if strings.ToLower(mp.Destination) == `c:\` {
-			return nil, fmt.Errorf(`Destination path in '%s' cannot be c:\`, spec)
-		}
-	}
-
-	// See if the source is a name instead of a host directory
-	if len(mp.Source) > 0 {
-		validName, err := IsVolumeNameValid(mp.Source)
-		if err != nil {
-			return nil, err
-		}
-		if validName {
-			// OK, so the source is a name.
-			mp.Name = mp.Source
-			mp.Source = ""
-
-			// Set the driver accordingly
-			mp.Driver = volumeDriver
-			if len(mp.Driver) == 0 {
-				mp.Driver = DefaultDriverName
-			}
-		} else {
-			// OK, so the source must be a host directory. Make sure it's clean.
-			mp.Source = filepath.Clean(mp.Source)
+	if source, exists := matchgroups["source"]; exists {
+		if source != "" {
+			split = append(split, source)
 		}
 	}
-
-	// Ensure the host path source, if supplied, exists and is a directory
-	if len(mp.Source) > 0 {
-		var fi os.FileInfo
-		var err error
-		if fi, err = os.Stat(mp.Source); err != nil {
-			return nil, fmt.Errorf("Source directory '%s' could not be found: %s", mp.Source, err)
-		}
-		if !fi.IsDir() {
-			return nil, fmt.Errorf("Source '%s' is not a directory", mp.Source)
+	if destination, exists := matchgroups["destination"]; exists {
+		if destination != "" {
+			split = append(split, destination)
 		}
 	}
-
-	// Fix #26329. If the destination appears to be a file, and the source is null,
-	// it may be because we've fallen through the possible naming regex and hit a
-	// situation where the user intention was to map a file into a container through
-	// a local volume, but this is not supported by the platform.
-	if len(mp.Source) == 0 && len(mp.Destination) > 0 {
-		var fi os.FileInfo
-		var err error
-		if fi, err = os.Stat(mp.Destination); err == nil {
```

Comment:
It is the right field. The regex doesn't match it as a name (correctly), so even though it really is a source, it ends up in destination. Hence why the trap a few lines up doesn't catch it